overlord/servicestate/quota_handlers.go

```diff
@@ -221,6 +219,80 @@ func addRestartServicesTasks(st *state.State, queueTask func(task *state.Task),
 	}
 }
 
+func (m *ServiceManager) doQuotaAddSnap(t *state.Task, _ *tomb.Tomb) error {
+	st := t.State()
+	st.Lock()
+	defer st.Unlock()
+
+	perfTimings := state.TimingsForTask(t)
+	defer perfTimings.Save(st)
+
+	snapsup, err := snapstate.TaskSnapSetup(t)
+	if err != nil {
+		return err
+	}
+
+	var quotaName string
+	if err := t.Get("quota-name", &quotaName); err != nil {
+		return fmt.Errorf("internal error: cannot get quota-name: %v", err)
+	}
+
+	allGrps, err := AllQuotas(st)
+	if err != nil {
+		return err
+	}
+
+	qc := QuotaControlAction{
+		Action:    "update",
+		QuotaName: quotaName,
+		AddSnaps:  []string{snapsup.InstanceName()},
+	}
+	grp, allGrps, _, err := quotaUpdate(st, qc, allGrps)
+	if err != nil {
+		return err
+	}
+
+	// ensure service and slices on disk and their states are updated
+	opts := &ensureSnapServicesForGroupOptions{
+		allGrps: allGrps,
+	}
+
+	servicesAffected, err := ensureSnapServicesForGroup(st, t, grp, opts)
+	if err != nil {
+		return err
+	}
+
+	// ensure that if any services are affected, they get their profiles
+	// refreshed immediately as a part of the install change.
+	if len(servicesAffected) > 0 && grp.JournalLimit != nil {
```

Comment:
If `grp.JournalLimit` is `nil`, does it mean that the limit has been removed? Shouldn't we then run the refresh-profile task even in that case, to lift the limits?

Reply:
In this specific task, we are not modifying any limits of the group, only adding snaps to a group, which means that we can trust the journal quota is not being changed for the duration of the task. So we only run the refresh-profiles if services were affected and the group has a journal limit